var/spack/repos/builtin/packages/py-pennylane/package.py

```diff
@@ -10,25 +10,45 @@
 class PyPennylane(PythonPackage):
     """PennyLane is a Python quantum machine learning library by Xanadu Inc."""
 
-    homepage = "https://github.com/XanaduAI/pennylane"
+    homepage = "https://docs.pennylane.ai/"
+    git = "https://github.com/PennyLaneAI/pennylane.git"
     pypi = "PennyLane/PennyLane-0.28.0.tar.gz"
 
-    maintainers("marcodelapierre")
+    maintainers = ["mlxd", "AmintorDusko"]
 
+    version("develop", branch="master")
     version("0.28.0", sha256="2a6100c00277c1eb59eab6856cdad7b1237e9d1fbda98b1e15020bd5a64b10a8")
 
     depends_on("python@3.8:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
-
-    depends_on("py-numpy@:1.23", type=("build", "run"))
-    depends_on("py-scipy", type=("build", "run"))
-    depends_on("py-networkx", type=("build", "run"))
-    depends_on("py-retworkx", type=("build", "run"))
-    depends_on("py-autograd", type=("build", "run"))
-    depends_on("py-toml", type=("build", "run"))
-    depends_on("py-appdirs", type=("build", "run"))
     depends_on("py-semantic-version@2.7:", type=("build", "run"))
-    depends_on("py-autoray@0.3.1:", type=("build", "run"))
-    depends_on("py-cachetools", type=("build", "run"))
-    depends_on("py-pennylane-lightning@0.28:", type=("build", "run"))
-    depends_on("py-requests", type=("build", "run"))
+    depends_on("py-pip", type="build")
+    depends_on("py-wheel", type="build")
+
+    depends_on("py-appdirs", type="run")
+    depends_on("py-autograd", type="run")
+    depends_on("py-autoray", type="run")
+    depends_on("py-cachetools", type="run")
+    depends_on("py-networkx", type="run")
+    depends_on("py-numpy@:1.23", type="run")
+    depends_on("py-pennylane-lightning+python", type="run")
+    depends_on("py-retworkx", type="run")
+    depends_on("py-requests", type="run")
+    depends_on("py-scipy", type="run")
+    depends_on("py-toml", type="run")
```

Comment:
These should be `type=("build", "run")`, otherwise the tests won't work because these dependencies won't exist in `PYTHONPATH`.

Reply:
Also, it's much quicker to review these if they are in the same order as in `setup.py` (as they were before)

Reply:
We try to maintain dependencies in alphabetical order internally, but happy to go with whatever is preferred here.

Happy to make the change to ("build, "run"). Thought, the tests seem to run fine post-install for us using `spack install --verbose --test=root py-pennylane`. Is there a preferred way for us to validate these when we are testing the package internally?

Reply:
There are two types of tests in Spack.

1. Install tests: `spack install --test=root foo`
2. Post-install tests: `spack test run foo`

Unless you add additional tests yourself, these both attempt to import all Python files (for a pure Python package). At install-time, only dependencies listed as `type="build"` will be in `PYTHONPATH`. For post-install tests, only dependencies listed as `type="run"` will be in `PYTHONPATH`.

Reply:
Ah, I understand, thanks. The above was only validated against option 1.

Will check and make the changes.